chain/ethereum/src/adapter.rs

```diff
@@ -212,10 +212,12 @@ pub(crate) struct EthereumLogFilter {
     /// Log filters can be represented as a bipartite graph between contracts and events. An edge
     /// exists between a contract and an event if a data source for the contract has a trigger for
     /// the event.
-    contracts_and_events_graph: GraphMap<LogFilterNode, (), petgraph::Undirected>,
+    /// Edges are of `bool` type and indicates when a trigger requires a transaction receipt.
+    contracts_and_events_graph: GraphMap<LogFilterNode, bool, petgraph::Undirected>,
 
-    // Event sigs with no associated address, matching on all addresses.
-    wildcard_events: HashSet<EventSignature>,
+    /// Event sigs with no associated address, matching on all addresses.
+    /// Maps to a boolean representing if a trigger requires a transaction receipt.
+    wildcard_events: HashMap<EventSignature, bool>,
```

Comment:
I don't think so, because then we would allow a possible state where the same signature could both require and not require a transaction receipt.

Also, searching for the `bool` would require iteration, which is not fast as indexing.